modules/renter/files.go

```diff
@@ -107,6 +108,29 @@ func (f *file) uploadProgress() float64 {
 	return 100 * (float64(uploaded) / float64(desired))
 }
 
+// redundancy returns the redundancy of the least redundant chunk. A file
+// becomes available when this redundancy is >= 1. Assumes that every piece is
+// unique within a file contract.
+func (f *file) redundancy() float64 {
+	if f.numChunks() == 0 {
+		build.Critical("cannot get redundancy of a file with 0 chunks")
+		return 0
+	}
```

Comment:
What guarantees are there that you will never call redundancy on a file with 0 chunks? Is it true that all files, even empty files, are going to have at least one chunk?